libraries/joomla/form/form.php

```
@@ -301,15 +301,14 @@ public function getField($name, $group = null, $value = null)
 	 * @return  mixed  The attribute value for the field.
 	 *
 	 * @since   11.1
+	 * @throws  RuntimeException
 	 */
 	public function getFieldAttribute($name, $attribute, $default = null, $group = null)
 	{
 		// Make sure there is a valid JForm XML document.
 		if (!($this->xml instanceof JXMLElement))
 		{
-			// TODO: throw exception.
-
-			return $default;
+			throw new RuntimeException('XML is not an instance of JXMLElement');
```

I would throw this as an UnexpectedValueException like this:

```
throw new UnexpectedValueException(sprint('%s::getFieldAttribute `xml` is not an instance of JXMLElement', get_class($this)));
```

That way it's easier for the developer to home in on where the error is.  Ditto for the other cases.